pyprojects/models.py

```diff
@@ -20,5 +22,51 @@ def _pytype(self, v):
             return v.isoformat()
         return v
 
-class Project(db.Model, DictSerializableMixin):
+class Project(db.Model, DictSerializableMixin, Resource):
     __tablename__ = 'projects'
+    required_fields = {
+        'id' : str,
+        'name' : unicode,
+        'description' : unicode,
+        'start_date' : date,
+        'manager': unicode,
+        'budget': float
+    }
+    optional_fields = {
+        'end_date' : date
+    }
+    record_key = 'projects'
+    path = '/api/project'
+    
+    @classmethod
+    def index(cls):
+        q = cls.query
+        if 'order_by' in request.args and request.args['order_by'] in cls.__table__.columns:
+            q = q.order_by(request.args['order_by'])
+        else:
+            q= q.order_by(cls.id)
+        records = [r.serialize() for r in q]
+        return jsonify({cls.record_key: records, "length": len(records)})
+
+
+class Budget(db.Model, DictSerializableMixin, Resource):
```

Comment:
@birdage

This is going to make building Python based APIs unbelievably easy. I mean, like WHOA easy.
http://img.photobucket.com/albums/v211/Cam_a01/GIFs/tumblr_lsx5vyCbQD1qkyz1oo1_500.gif